dlt/destinations/dataset/ibis_relation.py

```diff
@@ -0,0 +1,161 @@
+        target_dialect = DIALECT_MAP[destination_type]
+
+        # render sql directly if possible
+        if target_dialect not in TRANSPILE_VIA_MAP:
+            return ibis.to_sql(self._expression, dialect=target_dialect)
+
+        # here we need to transpile first
+        transpile_via = TRANSPILE_VIA_MAP[target_dialect]
+        sql = ibis.to_sql(self._expression, dialect=transpile_via)
+        sql = sqlglot.transpile(sql, read=transpile_via, write=target_dialect)[0]
+        return sql
+
+    @property
+    def columns_schema(self) -> TTableSchemaColumns:
+        return self.compute_columns_schema()
+
+    @columns_schema.setter
+    def columns_schema(self, new_value: TTableSchemaColumns) -> None:
+        raise NotImplementedError("columns schema in ReadableDBAPIRelation can only be computed")
+
+    def compute_columns_schema(self) -> TTableSchemaColumns:
+        """provide schema columns for the cursor, may be filtered by selected columns"""
+        # TODO: provide column lineage tracing with sqlglot lineage
+        return None
+
+    def _proxy_expression_method(self, method_name: str, *args: Any, **kwargs: Any) -> Any:
+        """Proxy method calls to the underlying ibis expression, allowing to wrap the resulting expression in a new relation"""
+
+        # Get the method from the expression
+        method = getattr(self._expression, method_name)
+
+        # unwrap args and kwargs if they are relations
+        args = tuple(
+            arg._expression if isinstance(arg, ReadableIbisRelation) else arg for arg in args
+        )
+        kwargs = {
+            k: v._expression if isinstance(v, ReadableIbisRelation) else v
+            for k, v in kwargs.items()
+        }
+
+        # casefold string params, we assume these are column names
+        args = tuple(
+            self.sql_client.capabilities.casefold_identifier(arg) if isinstance(arg, str) else arg
+            for arg in args
+        )
+        kwargs = {
+            k: self.sql_client.capabilities.casefold_identifier(v) if isinstance(v, str) else v
+            for k, v in kwargs.items()
+        }
+
+        # Call it with provided args
+        result = method(*args, **kwargs)
+
+        # If result is an ibis expression, wrap it in a new relation else return raw result
+        if isinstance(result, Expr):
+            return self.__class__(readable_dataset=self._dataset, expression=result)
+        return result
+
+    def __getattr__(self, name: str) -> Any:
+        """Wrap all callable attributes of the expression"""
+
+        attr = getattr(self._expression, name, None)
+
+        # try casefolded name for ibis columns access
+        if attr is None:
+            name = self.sql_client.capabilities.casefold_identifier(name)
+            attr = getattr(self._expression, name, None)
+
+        if attr is None:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{name}'")
+
+        if not callable(attr):
```

Comment:
I think ibis supports a column getter with a dot notation. my take would be to 
- start supporting it upstream
- modify the schema accordingly

Reply:
I'm not sure what you mean, the column getter dot notation is already working, see the tests, I'm using it there all the time for filtering and joining etc.